### PR TITLE
Updated regex to ignore backslash

### LINKED ISSUE #427: /import command is sensitive to trailing backslash
Compare the results of doing both of these:

1. `/import https://github.com/tarasglek/chatcraft.org/pull/421/` with trailing backslash
2. `https://github.com/tarasglek/chatcraft.org/pull/421` without trailing backslash

I think the issue here is that we don't want to mess with a URL like https://github.com/tarasglek/chatcraft.org/pull/421#issue-2124528458 (comment).

The code in question is in https://github.com/tarasglek/chatcraft.org/blob/main/functions/lib/rewriters/github-rewriter.ts

### PR DESCRIPTION
**Pull request by Rachit1313**
Closes #427 

### Changes Made:
Updated regex to ignore backslash in github commands.

### How to test:
Both the links with backslash at the end or without would recieve diff when a link of github PR is imported using /import command.

### CODE REVIEW

**Review by kliu57 — CHANGES_REQUESTED** [review]
```
Behvaiour is different for /import https://github.com/tarasglek/chatcraft.org/pull/421

Than /import http://github.com/tarasglek/chatcraft.org/pull/421
```

```
Moreover, behaviour is different from production for /import https://github.com/tarasglek/chatcraft.org/pull/421/ and all others

Output stops at line 83
```
image

**Comment by kliu57** [comment]
Cancelling this PR since the change was made in commit https://github.com/tarasglek/chatcraft.org/commit/bcdd39ca56ae7a6b73f1516833f3f1488ed6dbf3

File: functions/lib/rewriters/github-rewriter.ts [hunk 1]
```diff
@@ -8,11 +8,11 @@ export class GitHubRewriter extends DefaultRewriter {
 
   async rewriteUrl(url: URL) {
     // Blob URLs - https://github.com/<owner>/<repo>/blob/<branch>/<path>
-    if (url.origin === "https://github.com" && /\/(.*)\/blob\/(.*)$/.test(url.pathname)) {
-      // https://github.com/<owner>/<repo>/blob/<branch>/<path> -> https://raw.githubusercontent.com<owner>/<repo>/<branch>/<path>
+    if (url.origin === "https://github.com" && /\/(.*)\/blob\/(.*)\/?$/.test(url.pathname)) {
```
Comment (humphd) [inline functions/lib/rewriters/github-rewriter.ts:11]:
```
# Works
/import https://github.com/tarasglek/chatcraft.org/blob/main/.eslintignore

# Fails
/import https://github.com/tarasglek/chatcraft.org/blob/main/.eslintignore/
```

The latter returns a 400 from the server

File: functions/lib/rewriters/github-rewriter.ts [hunk 2]
```diff
@@ -23,14 +23,14 @@ export class GitHubRewriter extends DefaultRewriter {
...
-          /^https:\/\/gist\.github\.com\/([a-zA-Z0-9_-]+)\/([a-f0-9]+)$/,
+          /^https:\/\/gist\.github\.com\/([a-zA-Z0-9_-]+)\/([a-f0-9]+)\/?$/,
           "https://gist.githubusercontent.com/$1/$2/raw"
         )
       );
     }
 
     // PR URLs - https://github.com/<owner>/<repo>/pull/<number>
-    if (url.origin === "https://github.com" && /\/(.*)\/pull\/(.*)$/.test(url.pathname)) {
+    if (url.origin === "https://github.com" && /\/(.*)\/pull\/(.*)\/?$/.test(url.pathname)) {
```
Comment (humphd) [inline functions/lib/rewriters/github-rewriter.ts:33]: This one seems to work well for me.

File: functions/lib/rewriters/github-rewriter.ts [hunk 3]
```diff
@@ -23,14 +23,14 @@ export class GitHubRewriter extends DefaultRewriter {
       // https://gist.github.com/<owner>/<gist-id> -> https://gist.githubusercontent.com/<owner>/<gist-id>/raw
       return new URL(
         url.href.replace(
-          /^https:\/\/gist\.github\.com\/([a-zA-Z0-9_-]+)\/([a-f0-9]+)$/,
+          /^https:\/\/gist\.github\.com\/([a-zA-Z0-9_-]+)\/([a-f0-9]+)\/?$/,
```
Comment (humphd) [inline functions/lib/rewriters/github-rewriter.ts:26]: This is working well for me.

**Review by humphd — CHANGES_REQUESTED** [review]

File: functions/lib/rewriters/github-rewriter.ts (hunk 1 above)
Comment (Rachit1313) [reply functions/lib/rewriters/github-rewriter.ts:11]: I just verified that this is something we have in production as well

Comment (humphd) [reply functions/lib/rewriters/github-rewriter.ts:11]: Do you want to fix it while you're working on this or split it off into a future issue?

Comment (Rachit1313) [reply functions/lib/rewriters/github-rewriter.ts:11]: I would like to work on debugging this, but as this PR fixes the problem for PRs and gist, i would like to get this merged and create another issue and PR for that one.

**Comment by Rachit1313** [comment]
And for the second part i.e difference between https and http, I believe this is because we are only looking for https in the origin of the URL to confirm if we want to rewrite this with github-rewriters.ts. Everywhere in our code we used https to verify.
image
I am not sure, just guessing, @humphd would be able to answer this.

**Comment by humphd** [comment]
> @kliu57 , for me both the import command on PR 412 without backslash returns the same number of lines i.e 83 in production as well in my branch preview URL. image image

this is likely due to it being a diff, with only ~6 lines of context (so it naturally cuts if off), which is normal.

**Comment by humphd** [comment]
> @kliu57 And for the second part i.e difference between https and http, I believe this is because we are only looking for https in the origin of the URL to confirm if we want to rewrite this with github-rewriters.ts. Everywhere in our code we used https to verify. image I am not sure, just guessing, @humphd would be able to answer this.

I think requiring HTTPS is fine.

**Review by mingming-ma — COMMENTED** [review]
Here is comparing results:

| | Prod | This PR |
| ------------- | ------------- | ------|
| import https, has end slash| image  | image  |
| import https, no end slash  | image| image |
| import http, has end slash| image  |image  |
| import http, no  end slash  | image| image |

**Comment by Rachit1313** [comment]
I think we're good to land this. I'll fix for 400 error of blob as a separate issue.

**Review by rjwignar — APPROVED** [review]
These changes LGTM.
I think the Blob URL issue can be resolved in a followup.

**Pull request merged**